### PR TITLE
Testing Enumerables

### PR DESCRIPTION
**Pull request by ricardomonte**
In this project we:

- Use Rspec to test all of the methods in Enumerables 
- Apply different tests for each method

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+describe Enumerable do
+  let(:array) { [1, 2, 3, 4] }
+  let(:array2) { [2, 4, 6, 8] }
+  let(:range) { (1...5) }
+  let(:hash) { { 1 => 'one', 2 => 'two' } }
+  let(:animal) { %w[cat dog wombat] }
+  describe '#my_each' do
+    context 'with a block given' do
+      it 'return an array' do
+        expect(array.my_each { |num| puts num }).to eq(array.each { |num| puts num })
```
Comment (elmejdki) [inline spec/enumerable_spec.rb]: remove puts from the block giving to my_each and each.

File: README.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ Ruby
 
 You will be redirected to a webpage where you can provide a list of numbers or strings to be sorted when you run the code.
 
+## Testing
+
+-For Testing:
+
+    - Open the folder that contains enumerable.rb.
+    - from that folder open the terminal.
```
Comment (elmejdki) [inline README.md:32]: add another line that said that the user should run bundle install.

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Great job so far guys :clap: :clap: :clap: :clap: :clap:, but you still have just some minor changes :hammer: :wrench: :toolbox: to take care of.

- [ ] add test cases for multiply_els.

### Please make sure that you have worked on all the requirements before you submit another code review.
## Happy Coding :tophat:

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request reviewed by Maria Eugenia Queme P. (TSE)

### Hello @ricardomonte and @edxco :lollipop: :balloon: 

### Thank you for doing the changes requested by the previous reviewer :medal_sports: :medal_sports:  you've added several scenarios and your test cases have meaningful descriptions there are still mandatory changes so let's address them :construction: :construction_worker: :construction_worker: 

#### Project specific

- [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

> you seem to have positive and negative expected behaviors within the same `it` block, this is not correct each `it` block should contain one expected behavior, please separate them and use negative descriptions for the negative expected behaviors for example the negative of this it block 'Returns a new array with the results of running block once for every element in the collection.' could be "it doesn't return the original array when running a block once for every element in the collection."

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

### Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Review by kubilaycaglayan — APPROVED** [review]
Hello Team :wave: 
Status: Project Approved :green_circle: 

Thanks for making the required changes :+1: :blush: 

With all my respect :pray:

**Pull request merged**